modules/internal/DefaultRectangular.chpl

```diff
@@ -644,6 +644,13 @@ module DefaultRectangular {
     pragma "local field"
     var shiftedData : _ddata(eltType);
     var noinit_data: bool = false;
+
+    // 'dataAllocDom' is used by the array-vector operations (e.g. push_back,
+    // pop_back, insert, remove) to allow growing or shrinking the data
+    // buffer in a doubling/halving style.  If it is used, it will be the
+    // actual size of the data buffer, while 'dom' represents the size of the
+    // user-level array.
+    var dataAllocDom: domain(1);
```

Comment:
I just changed dataAllocDom into a range and am running it through paratest now.  I need it to track both a high and low bound, so a pair of integers would work, but a range's methods make it nicer.

We don't currently know statically if an array will be used as a vector in this way.  Any non-strided 1D Rectangular array with a unique domain (checked at runtime) can be used like this.